### ISSUE TITLE
Servers monitoring Akimbo map packages for lastest revisions maps and auto-update them.

### ISSUE THREAD

**Issue by AC-MPx** [assaultcube/AC#119]
NOTE: By request, moved to here.

Well, basically. Servers should monitoring the map on his 'servermaps' folder and update if there's a new version in one of the maps that have. For example:

The server "Custom maps" has a map (pipo_map) wich is already uploaded in akimbo and then the mapper upload a new version (to akimbo), the server "custom maps" should download and replace the map with the new version.

I think this will be useful, because the mapper doesn't have to upload the map in all the servers where is admin or on those servers where uploading maps are available for normal players. The same for servers owners.
This feature have to be server side and the server have to compare the map revision to /getmap automatically from akimbo, based on the maps that has on his 'servermaps' folder ofc.

Sorry for the noobish explanation.

**Comment by ac-stef** [comment]
That would be great. The problem is, that (afaik) the current media servers do not support versioning of any kind. So the server would have to download maps from akimbo to check, if they differ from the versions it already has. This would create a lot of unnecessary traffic for akimbo and the servers. We would need a way to retrieve version info from akimbo without downloading complete maps.

Btw, the exact same problem applies to custom media files (textures, mapmodels) - only in this case it's the client who doesn't know, if it should re-download certain media files.

**Comment by AC-MPx** [comment]
yeah, its beyond me. the idea it's great and could be used for the 'small updates' not only for maps. But for the game files.

About the custom media. I don't know. that its client-side. If the client doesn't have the custom media or the media changed, the auto-download will be in charge of that. because the entries are on the cfg

Probably this is useless: http://i.imgur.com/qgS1aeW.png
Akimbo need something like the auto-download but for map revision. Will notify the servers the new map rev, if the server doesn't have the map, will discard the notification.

**Comment by GreenLunar** [comment]
This is important for #123

**Comment by GreenLunar** [comment]
> Probably this is useless: http://i.imgur.com/qgS1aeW.png

Why AC_CLIENT also downloads from AKIMBO, if AC_SERVER can force AC_CLIENT to download older maps?

**Comment by GreenLunar** [comment]
oops
never mind
